src/ScreenshotManager.vala

```diff
@@ -97,8 +97,10 @@ namespace Gala
 			selection_area.start_selection ();
 
 			yield;
+			selection_area.destroy ();
+			
+			yield wait_stage_repaint ();
 			selection_area.get_selection_rectangle (out x, out y, out width, out height);
```

Comment:
You appear to be referencing selection area after destroying it?

Reply:
`destroy` doesn't unref the object itself, only frees all the resources allocated by Clutter for rendering and removes it from the stage, the object still exists. `selection_area` itself will be freed after the function ends.

Reply:
Ah, right.  That would explain why it doesn't crash!